feature_engine/encoding/decision_tree.py

```diff
@@ -182,6 +183,7 @@ def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
             check_classification_targets(y)
 
         # check input dataframe
+        X, y = _check_X_y_pd_np_mismatch(X, y)
```

Comment:
Do we need this for this encoder? because X and y are being passed to cross_validate, and that should call for sklearn machinery to take care of X and y.

Reply:
The test_detect_index_mismatch_from_x_pandas_y_pandas() unit test fails without it. I think it has to do with DecisionTreeEncoder using OrdinalEncoder but not sure. At any rate this call seems to be needed here to pass unit tests.